compiler/hash-lower/src/build/ty.rs

```diff
@@ -0,0 +1,212 @@
+                                _ => {
+                                    // if the fields of the struct are not opaque, then we
+                                    // can create an ADT from it, otherwise this case should
+                                    // not occur, and we have encountered an unhandled primitive.
+                                    if let StructFields::Explicit(params) = struct_def.fields {
+                                        let fields = tcx
+                                            .params_store
+                                            .get_vec(params)
+                                            .into_iter()
+                                            .enumerate()
+                                            .map(|(index, param)| AdtField {
+                                                name: param.name.unwrap_or_else(|| index.into()),
+                                                ty: convert_term_into_ir_ty(param.ty, tcx, ir_ctx),
+                                            })
+                                            .collect();
+
+                                        let variants = index_vec![AdtVariant { name, fields }];
+
+                                        let adt = AdtData::new_with_flags(
+                                            name,
+                                            variants,
+                                            AdtFlags::STRUCT,
+                                        );
+                                        let adt_id = ir_ctx.adt_store().create(adt);
+
+                                        IrTy::Adt(adt_id)
+                                    } else {
+                                        // If we get here, this means that we haven't accounted for
+                                        // a particular primitive
+                                        // type occurring.
+                                        panic!("unhandled primitive type: {name}");
+                                    }
+                                }
+                            }
+                        } else {
+                            todo!()
+                        }
+                    }
+                    NominalDef::Unit(_) => todo!(),
+
+                    // @@Remove: this will later be removed, so don't deal with this case for now.
+                    NominalDef::Enum(_) => unreachable!(),
+                }
+            }
+            Level1Term::Tuple(TupleTy { members }) => {
+                let params = tcx
+                    .params_store
+                    .get_owned_param_list(members)
+                    .into_positional()
+                    .into_iter()
+                    .map(|param| convert_term_into_ir_ty(param.ty, tcx, ir_ctx));
+
+                let params_id = ir_ctx.ty_list_store().create_from_iter(params);
+                IrTy::Tuple(params_id)
+            }
```

Comment:
Maybe directly replace with adt here?

Reply:
Yes, ill add it as another variant of the ADT